File: CommonDomain/src/main/java/domain/Entity.java
package domain;

import java.io.Serializable;

public class Entity <E> implements Serializable {
    private E id;

    public Entity(E id) {
        this.id=id;
    }

    public Entity(){}

    public E getID() { return id;}
    public void setID(E id) { this.id=id;}

}

File: CommonDomain/Entity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CommonDomain
{
    [Serializable]
    public class Entity<TE>
    {
        private TE Id;

        protected Entity() { }
        protected Entity(TE newId)
        {
            this.Id = newId;
        }

        public TE GetId()
        {
            return this.Id;
        }

        public void SetId(TE newId)
        {
            this.Id = newId;
        }
    }
}